third_party/move/move-compiler/src/cfgir/cfg.rs

```diff
@@ -192,6 +192,72 @@ impl<'a> BlockCFG<'a> {
             println!();
         }
     }
+
+    /// Adds a block on any critical edge: an edge from a node with multiple successors to
+    /// a node with multiple predecessors.  Returns true if any changes occur; in this
+    /// case, caller must call recompute().
+    pub fn remove_critical_edges<Ftype>(&mut self, mut make_label: Ftype) -> bool
+    where
+        Ftype: FnMut() -> Label,
+    {
+        let critical_edges: Vec<_> = self
```

Comment:
I don't think you need to construct the vector, you just can use the iterator in the for loop below. (Clippy should(?) tell you if I'm right -- did you run `rust_lint`?)

Reply:
There's a borrow problem with that.  To get the items I need &self, but to split edges I need a &mut self.  Easier to leave it separate.  I've changed slightly it to consume critical_edges in the loop.